src/orders/views/OrderTransactionRefundCreate/OrderTransactionRefundCreate.tsx

```diff
@@ -56,11 +57,19 @@ const OrderTransactionRefund: React.FC<OrderTransactionRefundCreateProps> = ({ o
         orderId,
         amount,
         reason,
-        lines: linesToRefund.map(line => ({
-          quantity: line.quantity,
-          reason: line.reason,
-          id: data.order!.lines[line.row].id,
-        })),
+        lines: linesToRefund.reduce<OrderGrantRefundCreateLineInput[]>((acc, line, ix) => {
```

Comment:
Suggestion: filter out all strings then run reduce. Since we're making array out of `linesToRefund` then `reduce` may be replaced with `filter` and/or `map`.

Reply:
plus - it deserves a dedicated function, please don't make big components even bigger